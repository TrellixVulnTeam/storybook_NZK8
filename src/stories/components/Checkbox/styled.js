// @flow

import styled from 'styled-components';
import { colors } from 'utils/StyleGuide';

export const DisabledWrapper = styled.div`
  position: relative;
  width: 16px;
  height: 16px;
  display: block;
  margin: 0 auto;
  font-size: 14px;
  border-radius: 2px;
  border: 1px solid ${colors.paleGrey};
  background-color: ${colors.lightGrey};
  transition: all 100ms ease;
  cursor: default;
`;

export const CheckboxWrapper = styled.div`
  display: flex;
  align-items: center;
`;

export const CheckboxElem = styled.div`
  position: relative;
  width: 16px;
  height: 16px;
  display: ${p => (p.label ? 'inline-block' : 'block')};
  margin: ${p => (p.label ? 0 : '0 auto')};
  font-size: 14px;
  border-radius: 2px;
  border: 1px solid ${p => (p.checked ? colors.transparent : colors.silver)};
  background-color: ${p => (p.disabled ? colors.silver : colors.white)};
  background-image: ${p => p.checked && 'linear-gradient(223deg, #00aeff, #0076ff)'};
  transition: all 100ms ease;
  cursor: pointer;

  &:hover {
    background-image: ${p => p.checked && 'none'};
    background-color: ${p => (p.checked ? colors.white : colors.iceBlue)};
    border-color: ${colors.brightBlue};

    > span {
      border-color: ${colors.brightBlue};
    }
  }

  /* checkmark */
  span {
    display: ${p => (p.checked ? 'block' : 'none')};
    position: absolute;
    top: 1px;
    left: 5px;
    width: 5px;
    height: 10px;
    border: solid ${colors.white};
    border-width: 0 2px 2px 0;
    transform: rotate(45deg);
    transition: all 100ms ease;
  }
`;

export const Label = styled.label`
  display: inline-block;
  ${({ labelWidth }) =>
    labelWidth
      ? `width: ${labelWidth}px
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;`
      : null};
  margin-left: 10px;
  cursor: pointer;
`;
