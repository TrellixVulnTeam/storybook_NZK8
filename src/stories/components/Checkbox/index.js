// @flow

import React from 'react';
import Tooltip from 'components/Tooltip';
import Asterisk from 'components/RequiredAsterisk';
import { CheckboxWrapper, CheckboxElem, DisabledWrapper, Label } from './styled';

type Props = {
  onClick?: Function,
  checked?: *,
  disabled?: boolean,
  tooltipMessage?: string,
  id?: string,
  label?: string,
  // TODO: Chech these props usage @moidmw
  // title?: string,
  // labelWidth?: number,
  className?: string,
  required?: boolean,
};

export default (props: Props) => {
  if (props.disabled) {
    if (props.tooltipMessage) {
      return (
        <Tooltip
          value={props.tooltipMessage}
          place="right"
          style={{ width: '100%' }}
          id={props.id && props.id}
        >
          <DisabledWrapper onClick={props.onClick} />
        </Tooltip>
      );
    }
    return <DisabledWrapper />;
  }

  return (
    <CheckboxWrapper className={props.className}>
      <CheckboxElem label={props.label} onClick={props.onClick} checked={props.checked}>
        <span />
      </CheckboxElem>
      {props.label && <Label onClick={props.onClick}>{props.label}</Label>}
      {props.required && <Asterisk />}
    </CheckboxWrapper>
  );
};
