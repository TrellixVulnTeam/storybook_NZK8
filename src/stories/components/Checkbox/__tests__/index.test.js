import React from 'react';
import { mount } from 'enzyme';
import Checkbox from 'components/Checkbox/index';

describe('<Checkbox />', () => {
  it('should render correctly with tooltip', () => {
    const props = {
      onClick: jest.fn(),
      disabled: true,
      tooltipMessage: 'Tooltip',
      id: '0',
    };
    const sut = mount(<Checkbox {...props} />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly without tooltip', () => {
    const props = {
      onClick: jest.fn(),
      disabled: true,
    };
    const sut = mount(<Checkbox {...props} />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly if enabled', () => {
    const props = {
      onClick: jest.fn(),
      checked: true,
      disabled: false,
      label: 'label',
    };
    const sut = mount(<Checkbox {...props} />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly if enabled without label and check', () => {
    const props = {
      onClick: jest.fn(),
      disabled: false,
    };
    const sut = mount(<Checkbox {...props} />);

    expect(sut).toMatchSnapshot();
  });
});
