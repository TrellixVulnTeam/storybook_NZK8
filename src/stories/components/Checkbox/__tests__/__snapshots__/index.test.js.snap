// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> should render correctly if enabled 1`] = `
<_default
  checked={true}
  disabled={false}
  label="label"
  onClick={[MockFunction]}
>
  <styled__CheckboxWrapper>
    <div
      className="styled__CheckboxWrapper-sc-1sf4nv3-1 kdBfUE"
    >
      <styled__CheckboxElem
        checked={true}
        label="label"
        onClick={[MockFunction]}
      >
        <div
          checked={true}
          className="styled__CheckboxElem-sc-1sf4nv3-2 cxsapr"
          label="label"
          onClick={[MockFunction]}
        >
          <span />
        </div>
      </styled__CheckboxElem>
      <styled__Label
        onClick={[MockFunction]}
      >
        <label
          className="styled__Label-sc-1sf4nv3-3 NmbMZ"
          onClick={[MockFunction]}
        >
          label
        </label>
      </styled__Label>
    </div>
  </styled__CheckboxWrapper>
</_default>
`;

exports[`<Checkbox /> should render correctly if enabled without label and check 1`] = `
<_default
  disabled={false}
  onClick={[MockFunction]}
>
  <styled__CheckboxWrapper>
    <div
      className="styled__CheckboxWrapper-sc-1sf4nv3-1 kdBfUE"
    >
      <styled__CheckboxElem
        onClick={[MockFunction]}
      >
        <div
          className="styled__CheckboxElem-sc-1sf4nv3-2 efVoBK"
          onClick={[MockFunction]}
        >
          <span />
        </div>
      </styled__CheckboxElem>
    </div>
  </styled__CheckboxWrapper>
</_default>
`;

exports[`<Checkbox /> should render correctly with tooltip 1`] = `
<_default
  disabled={true}
  id="0"
  onClick={[MockFunction]}
  tooltipMessage="Tooltip"
>
  <_default
    id="0"
    place="right"
    style={
      Object {
        "width": "100%",
      }
    }
    value="Tooltip"
  >
    <styled__Wrapper
      data-for="0"
      data-tip="Tooltip"
      id="0"
      place="right"
      style={
        Object {
          "width": "100%",
        }
      }
      value="Tooltip"
    >
      <div
        className="styled__Wrapper-funpqb-0 cznzQQ"
        data-for="0"
        data-tip="Tooltip"
        id="0"
        style={
          Object {
            "width": "100%",
          }
        }
        value="Tooltip"
      >
        <styled__DisabledWrapper
          onClick={[MockFunction]}
        >
          <div
            className="styled__DisabledWrapper-sc-1sf4nv3-0 vdKVP"
            onClick={[MockFunction]}
          />
        </styled__DisabledWrapper>
        <styled__TooltipWrapper
          display="true"
        >
          <div
            className="styled__TooltipWrapper-funpqb-1 gQtIVd"
            display="true"
          >
            <ReactTooltip
              clickable={false}
              delayShow={300}
              effect="solid"
              getContent={[Function]}
              html={true}
              id="0"
              insecure={true}
              multiline={true}
              place="right"
              resizeHide={true}
              type="dark"
              wrapper="div"
            >
              <div
                className="__react_component_tooltip place-right type-dark"
                data-id="tooltip"
                id="0"
              >
                null
              </div>
            </ReactTooltip>
          </div>
        </styled__TooltipWrapper>
      </div>
    </styled__Wrapper>
  </_default>
</_default>
`;

exports[`<Checkbox /> should render correctly without tooltip 1`] = `
<_default
  disabled={true}
  onClick={[MockFunction]}
>
  <styled__DisabledWrapper>
    <div
      className="styled__DisabledWrapper-sc-1sf4nv3-0 vdKVP"
    />
  </styled__DisabledWrapper>
</_default>
`;
